libs/ui-components/src/hooks/useDeviceLabelMatch.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { Icon, Spinner } from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import { ExclamationCircleIcon } from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+
+import { Fleet, FleetList } from '@flightctl/types';
+import { FlightCtlLabel } from '../types/extraTypes';
+import { useFetch } from './useFetch';
+import { useTranslation } from './useTranslation';
+import { getErrorMessage } from '../utils/error';
+import { CheckCircleIcon } from '@patternfly/react-icons/dist/js/icons';
+
+export type DeviceMatchStatus = {
+  status:
+    | 'unchecked'
+    | 'unchecked--invalid'
+    | 'checking'
+    | 'checked--error'
+    | 'checked--unique'
+    | 'checked--multiple'
+    | 'checked--empty';
+  detail?: string;
+};
+
+const getMatchResult = (fleets: Fleet[], deviceLabels: FlightCtlLabel[]): DeviceMatchStatus => {
+  const matchingFleets = fleets.filter((fleet) => {
+    const fleetMatch = fleet.spec.selector?.matchLabels || {};
+    return Object.entries(fleetMatch).every(([fleetMatchKey, fleetMatchValue]) => {
+      const matchingDeviceLabel = deviceLabels.find((dLabel) => dLabel.key === fleetMatchKey);
+      return matchingDeviceLabel && (matchingDeviceLabel.value || '') === (fleetMatchValue || '');
+    });
+  });
+
+  const matchCount = matchingFleets?.length || 0;
+
+  switch (matchCount) {
+    case 0:
+      return { status: 'checked--empty' };
+    case 1:
+      return { status: 'checked--unique', detail: matchingFleets[0].metadata.name };
+    default:
+      return { status: 'checked--multiple' };
+  }
+};
+
+const DeviceLabelMatchContent = ({
+  iconStatus,
+  text,
+  isPlaceholder,
+}: {
+  iconStatus?: 'success' | 'warning' | 'danger' | 'loader';
+  text?: string;
+  isPlaceholder?: boolean;
+}) => {
+  let icon: React.ReactNode;
+  switch (iconStatus) {
+    case 'warning':
+      icon = (
+        <Icon status="warning">
+          <ExclamationTriangleIcon />
+        </Icon>
+      );
+      break;
+    case 'danger':
+      icon = (
+        <Icon status="danger">
+          <ExclamationCircleIcon />
+        </Icon>
+      );
+      break;
+    case 'success':
+      icon = (
+        <Icon status="success">
+          <CheckCircleIcon />
+        </Icon>
+      );
+      break;
+    case 'loader':
+      icon = <Spinner size="sm" />;
+      break;
+  }
+
+  return (
+    <div className="fctl_approve-device__fleetname">
+      {icon && <span className="fctl_approve-device__fleetname__icon">{icon}</span>}
+      <span className={isPlaceholder ? 'fctl_approve-device__fleetname__placeholder' : ''}>{text}</span>
+    </div>
```

Comment:
I think we could reduce the amount of custom css by using PF's `Flex` 
```suggestion
    <Flex alignItems={{ default: 'alignItemsCenter' }} rowGap={{ default: 'rowGapSm' }}>
      {icon && <FlexItem>{icon}</FlexItem>}
      <FlexItem className={isPlaceholder ? 'fctl-approve-device__fleetname-placeholder' : ''}>{text}</FlexItem>
    </Flex>
```

Reply:
The problem I was having with the `Flex` component is with the error which takes more than 1 line.
I tried to figure out why the wrapping wasn't working properly, but couldn't find the combination of rules that provokes that and opted for using the custom CSS (which in theory wants to apply the same rules as in your example).

![no-wrap](https://github.com/user-attachments/assets/ab0f9193-15b2-45e3-b11a-7dc141eb6143)